### PR TITLE
Initialize refactoring

### PR DESCRIPTION
**Pull request by hobu**

### CODE REVIEW

**Comment by hobu** [comment]
Ok, this PR does a bunch of stuff:

* Any configuration metadata we want to i/o through the Storage class must go through the Configuration member
* Bounds transits via json
* CRS transits via json
* Bounds can be ctor'd via three common string spellings
* CRS CLI instantiation and validation
* Bounds CLI instantiation and validation

File: src/silvimetric/bounds.py [hunk 1]
```diff
@@ -1,16 +1,55 @@
...
         self.miny = float(miny)
         self.maxx = float(maxx)
         self.maxy = float(maxy)
 
+    def __eq__(self, other):
+        return  other.minx == self.minx and \
+                other.miny == self.miny and \
+                other.maxx == self.maxx and \
+                other.maxy == self.maxy
+    def __neq__(self, other):
```
Comment (kylemann16) [inline src/silvimetric/bounds.py:16]: I think this needs to be `__ne__()` to make the warning go away

Comment (hobu) [reply src/silvimetric/bounds.py]: You'll note I subclassed `dict` there to get free json serialization. 🤷 if that's the "right" thing to do but it was expedient.

File: src/silvimetric/storage.py [hunk 2]
```diff
@@ -60,78 +54,62 @@ def create(atts:list[str], resolution: float, bounds: list[float],
...
         return s
 
     def consolidate(self, ctx=None):
         # if not ctx:
         #     ctx = self.ctx
-        tiledb.consolidate(self.tdb_dir)
+        tiledb.consolidate(self.config.tdb_dir)
 
-    def saveMetadata(self, metadata: dict) -> None:
+    def saveConfig(self) -> None:
```
Comment (kylemann16) [inline src/silvimetric/storage.py:90]: Should we leave a `saveMetadata` method for when we're adding other metadata attributes to the db? In shatter I'm planning on adding at least `point_count` as a metadata item. Could just add manually with the tiledb write stream, but might be better to handle it in the storage class

Comment (kylemann16) [reply src/silvimetric/storage.py:90]: Sounds good to me 👍🏻

**Pull request merged**